### ISSUE TITLE
Build w/OpenBLAS fails, cannot find lapacke.h

### ISSUE THREAD

**Issue by iskunk** [kaldi-asr/kaldi#3440]
I am building Kaldi using OpenBLAS, and am encountering the following error under `src/`:
```
[...]
make -C matrix
make[1]: Entering directory '/tmp/kaldi-git/src/matrix'
g++ -std=c++11 -I.. -isystem /tmp/kaldi-git/tools/openfst-1.6.7/include -O1  -Wall -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-declarations -Winit-self -DKALDI_DOUBLEPRECISION=0 -DHAVE_EXECINFO_H=1 -DHAVE_CXXABI_H -DHAVE_OPENBLAS -I/tmp/kaldi-git/tools/OpenBLAS/install/include -msse -msse2 -pthread -g    -c -o kaldi-matrix.o kaldi-matrix.cc
In file included from ../matrix/cblas-wrappers.h:29:0,
                 from ../matrix/jama-svd.h:34,
                 from kaldi-matrix.cc:27:
../matrix/kaldi-blas.h:99:12: fatal error: lapacke.h: No such file or directory
   #include "lapacke.h"
            ^~~~~~~~~~~
compilation terminated.
<builtin>: recipe for target 'kaldi-matrix.o' failed
make[1]: *** [kaldi-matrix.o] Error 1
make[1]: Leaving directory '/tmp/kaldi-git/src/matrix'
Makefile:136: recipe for target 'matrix' failed
make: *** [matrix] Error 2
```

I see that `lapacke.h` is present in the OpenBLAS source tree:
```
$ find /tmp/kaldi-git -name lapacke.h
/tmp/kaldi-git/tools/OpenBLAS/lapack-netlib/LAPACKE/include/lapacke.h
```

However, it is not present in the OpenBLAS install tree, which is as follows:
```
$ tree -F /tmp/kaldi-git/tools/OpenBLAS/install
/tmp/kaldi-git/tools/OpenBLAS/install
|-- bin/
|-- include/
|   |-- cblas.h
|   |-- f77blas.h
|   `-- openblas_config.h
`-- lib/
    |-- cmake/
    |   `-- openblas/
    |       |-- OpenBLASConfig.cmake
    |       `-- OpenBLASConfigVersion.cmake
    |-- libopenblas.a -> libopenblas_nehalem-r0.3.5.a
    |-- libopenblas.so -> libopenblas_nehalem-r0.3.5.so*
    |-- libopenblas.so.0 -> libopenblas_nehalem-r0.3.5.so*
    |-- libopenblas_nehalem-r0.3.5.a
    |-- libopenblas_nehalem-r0.3.5.so*
    `-- pkgconfig/
        `-- openblas.pc

6 directories, 11 files
```

(I get the same tree if I build OpenBLAS independently, so this set of files appears to be correct.)

Is it correct for Kaldi to require `lapacke.h` when using OpenBLAS? It seems that OpenBLAS itself uses netlib/LAPACKE, but an OpenBLAS installation does not provide LAPACKE. Perhaps the build logic for OpenBLAS should additionally build netlib/LAPACKE for Kaldi's direct use?

**Comment by danpovey** [comment]
I just tested installing OpenBLAS using the provided script, and it did
include that file.  Perhaps try removing OpenBLAS and reinstalling it.

**Comment by iskunk** [comment]
I performed the following steps, under `tools/`:
```
$ rm -rf OpenBLAS

$ make openblas
[...]

$ tree -F OpenBLAS/install 
OpenBLAS/install
|-- bin/
|-- include/
|   |-- cblas.h
|   |-- f77blas.h
|   `-- openblas_config.h
`-- lib/
    |-- cmake/
    |   `-- openblas/
    |       |-- OpenBLASConfig.cmake
    |       `-- OpenBLASConfigVersion.cmake
    |-- libopenblas.a -> libopenblas_nehalem-r0.3.5.a
    |-- libopenblas.so -> libopenblas_nehalem-r0.3.5.so*
    |-- libopenblas.so.0 -> libopenblas_nehalem-r0.3.5.so*
    |-- libopenblas_nehalem-r0.3.5.a
    |-- libopenblas_nehalem-r0.3.5.so*
    `-- pkgconfig/
        `-- openblas.pc

6 directories, 11 files
```

Is there a different build script (aside from the makefile) I should be using?

**Comment by danpovey** [comment]
That's how you do it, yes; it's just not what I get when I do it, in my
case those files are included.
Attach the full output of those commands; and look for '**' in the make
output, to check for errors.

**Comment by iskunk** [comment]
I performed the following steps:
```
$ rm -rf OpenBLAS/

$ make openblas >make.out 2>&1

$ fgrep '**' make.out
(no output)

$ ls OpenBLAS/install/include/
cblas.h  f77blas.h  openblas_config.h
```

There doesn't appear to be any issue building OpenBLAS. In case there is anything I've missed, I am attaching the build log [here](https://github.com/kaldi-asr/kaldi/files/3347422/make.out.txt).

FWIW...
```
$ ./extras/check_dependencies.sh 
./extras/check_dependencies.sh: all OK.
```

**Comment by danpovey** [comment]
It says this at the top:
`OpenBLAS: Detecting fortran compiler failed. Cannot compile LAPACK. Only compile BLAS.`
We should update the make script to detect that, i.e. to fail if it can't find lapacke.h in the include dir.

**Comment by iskunk** [comment]
Oh, a Fortran compiler is required? I thought that would have been covered in `check_dependencies.sh`, etc.

**Comment by danpovey** [comment]
OpenBLAS is not the default, so it's not required normally.  We should put the check in an install script for OpenBLAS, and have the `make` target invoke that script.  @jtrmal @galv any chance one of you could work on it?

**Comment by danpovey** [comment]
Resolved in #3622
